File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/pool/basicfunctionality/item/ObjectsMaxCountReachedException.java
package ru.urfu.javapools.poolslibrary.pool.basicfunctionality.item;

import ru.urfu.javapools.poolslibrary.controller.PoolException;

@SuppressWarnings("serial")
public class ObjectsMaxCountReachedException extends PoolException {

	private int _maxObjectCount;
	
	private static final String _MESSAGE_PATTERN = "Object with key='%s' wasn't created because" +
			  									   " max objects count %s is already reached";
	
	public ObjectsMaxCountReachedException (Object key, int maxObjectCount) {
		super(key,String.format(_MESSAGE_PATTERN, key, maxObjectCount));
		_maxObjectCount = maxObjectCount;
	}
	
	public int getMaxObjectCount() {
		return _maxObjectCount;
	}
}

File: CSharpPools/PoolsLibrary/Pool/BasicFunctionality/Item/ObjectsMaxCountReachedException.cs
using PoolsLibrary.Controller;

namespace PoolsLibrary.Pool.BasicFunctionality.Item
{
    public class ObjectsMaxCountReachedException<TK> : PoolException<TK>
    {
        public int MaxObjectsCount { get; private set; }

        public ObjectsMaxCountReachedException(TK key, int maxObjectsCount, string message = null)
            : base(key, message)
        {
            MaxObjectsCount = maxObjectsCount;
        }
    }
}